transport.go

```diff
@@ -144,7 +144,7 @@ func getTransport(opts TransportOptions, encoding encoding.Encoding, tracer open
 			TargetService: opts.ServiceName,
 			HostPorts:     opts.HostPorts,
 			Encoding:      encoding.String(),
-			TransportOpts: opts.TransportOptions,
+			TransportOpts: opts.Headers,
```

Comment:
TransportOpts on this line is part of the transportutils interface, so not flexible on this end. opts is a TransportOptions object locally, TransportOptions.TransportHeaders would taste redundant.